File: src/main/java/lv/ldn/java8/NonFinal.java
package lv.ldn.java8;

import java.util.Arrays;
import java.util.List;
import java.util.concurrent.atomic.LongAdder;

public class NonFinal {

    public static void mutable() {
        int count = 0;
        List<String> strings = Arrays.asList("a", "b", "c");
        strings.forEach(s -> {
            // count++;  -- won't compile
        });
    }

    public static void atomic() {
        LongAdder count = new LongAdder();
        List<String> strings = Arrays.asList("a", "b", "c");
        strings.forEach(s -> count.increment());
    }

}

File: src/main/scala/lv/ldn/scala/NonFinal.scala
package lv.ldn.scala

import java.util.concurrent.atomic.LongAdder

object NonFinal {
  def mutable {
    var count = 0
    val strings = Seq("a", "b", "c")
    strings.foreach(s => count += 1)
  }

  def atomic {
    val count = new LongAdder
    val strings = Seq("a", "b", "c")
    strings.foreach(s => count.increment())
  }
}
